### PR TITLE
Make the mode an enum

### PR DESCRIPTION
**Pull request by jonnius**
How about making the mode an enum in order to enforce exactly one including proper error message by clap?

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -153,24 +166,24 @@ async fn start_tauri() {
...
+    use std::process::Stdio;
 
     log::info!("Starting the ubuntu touch client");
-    thread::spawn(|| {
-        let rt = tokio::runtime::Runtime::new().unwrap();
-        rt.block_on(async {
-            let route = warp::fs::dir("./axolotl-web/dist");
-            warp::serve(route).run(([127, 0, 0, 1], 9081)).await;
-        });
+    // Why do we start warp another time?
```
Comment (jonnius) [inline src/main.rs:172]: Why do we start warp another time?

Comment (nanu-c) [reply src/main.rs]: if we use ubuntu touch, the ui html/js has to be served via a file server. In tauri, the code is bundled with tauri. Actually we need another file server for the attachments.
It's not necessary to start another server, but we could add it as an route to warp if the feature ut is used.

Comment (jonnius) [reply src/main.rs]: understood

**Comment by jonnius** [comment]
I rebased and updated the PR. Feel free to close this if you don't like the enum approach.

File: src/main.rs [hunk 2]
```diff
@@ -52,123 +53,138 @@ async fn start_websocket(handler: Arc<Mutex<Handler>>) {
 
     warp::serve(axolotl_route).run(([127, 0, 0, 1], 9080)).await;
     log::info!("Server stopped");
-    // Why do we want to lock it shortly on shutdown?
-    let _ = handler_mutex.lock().await;
```
Comment (jonnius) [inline src/main.rs:56]: I hope there was no secret reason behind this line...

Comment (nanu-c) [reply src/main.rs:56]: lets try :)

**Pull request merged**